File: app/src/main/java/mega/privacy/android/app/MegaContactAdapter.java
package mega.privacy.android.app;


import mega.privacy.android.domain.entity.Contact;
import nz.mega.sdk.MegaUser;

public class MegaContactAdapter {

    Contact contact;
    MegaUser megaUser;
    String fullName;
    String lastGreen;
    private boolean isSelected;

    public MegaContactAdapter(Contact contact, MegaUser megaUser, String fullName) {
        this.contact = contact;
        this.megaUser = megaUser;
        this.fullName = fullName;
        this.lastGreen = "";
        this.isSelected = false;
    }

    public Contact getMegaContactDB() {
        return contact;
    }

    public void setMegaContactDB(Contact contact) {
        this.contact = contact;
    }

    public MegaUser getMegaUser() {
        return megaUser;
    }

    public void setMegaUser(MegaUser megaUser) {
        this.megaUser = megaUser;
    }

    public String getFullName() {
        return fullName;
    }

    public void setFullName(String fullName) {
        this.fullName = fullName;
    }

    public String getLastGreen() {
        return lastGreen;
    }

    public void setLastGreen(String lastGreen) {
        this.lastGreen = lastGreen;
    }

    public boolean isSelected() {
        return isSelected;
    }

    public void setSelected(boolean isSelected) {
        this.isSelected = isSelected;
    }
}

File: app/src/main/java/mega/privacy/android/app/MegaContactAdapter.kt
package mega.privacy.android.app

import mega.privacy.android.domain.entity.Contact
import nz.mega.sdk.MegaUser

/**
 * A data model represents the Mega Contact.
 *
 * @property contact [Contact]
 * @property megaUser [MegaUser]
 * @property fullName The full name of the contact
 * @property lastGreen Represents the time when the user last online
 * @property isSelected True if the contact is selected, false otherwise
 */
data class MegaContactAdapter @JvmOverloads constructor(
    val contact: Contact?,
    val megaUser: MegaUser?,
    val fullName: String?,
    var lastGreen: String = "",
    var isSelected: Boolean = false,
)
